insights/combiners/rhel_for_edge.py

```diff
@@ -58,13 +59,20 @@ class RhelForEdge(object):
 
     """
 
-    def __init__(self, rpms, cmdline, units, redhatrelease):
+    def __init__(self, units, rpmostreestatus, rpms, cmdline, redhatrelease):
         self.is_edge = False
         self.is_automated = False
-
-        if ('rpm-ostree' in rpms and 'yum' not in rpms) and \
-                ('ostree' in cmdline) and \
-                ("red hat enterprise linux release" in redhatrelease.raw.lower()):
-            self.is_edge = True
-            if units.is_running("rhcd.service"):
-                self.is_automated = True
+        if rpmostreestatus:
+            origin = rpmostreestatus.query.deployments.origin
+            origin_check = [item.value.endswith("edge") for item in origin]
+            if origin_check and all(origin_check):
+                self.is_edge = True
+                if units.is_running("rhcd.service"):
+                    self.is_automated = True
+        elif rpms and cmdline and redhatrelease:
+            if ('rpm-ostree' in rpms and 'yum' not in rpms) and \
+                    ('ostree' in cmdline) and \
+                    ("red hat enterprise linux release" in redhatrelease.raw.lower()):
+                self.is_edge = True
+                if units.is_running("rhcd.service"):
+                    self.is_automated = True
```

Comment:
Better add one more `else` at the end to handle (Skip) whenever any one of the `rpms and cmdline and redhatrelease` is missing.
And test it.

Reply:
@xiangce - I updated code to raise SkipComponent when `rpmostreestatus` or any of the `[rpms` , `cmdline`, redhatrelease]` is missing.